beacon_node/network/src/sync/backfill_sync/mod.rs

```diff
@@ -900,13 +900,14 @@ impl<T: BeaconChainTypes> BackFillSync<T> {
                     (
                         failed_peers.contains(peer),
                         self.active_requests.get(peer).map(|v| v.len()).unwrap_or(0),
+                        rand::thread_rng().gen::<u32>(),
                         *peer,
                     )
                 })
                 .collect::<Vec<_>>();
             // Sort peers prioritizing unrelated peers with less active requests.
             priorized_peers.sort_unstable();
-            priorized_peers.first().map(|&(_, _, peer)| peer)
+            priorized_peers.first().map(|&(_, _, _, peer)| peer)
```

Comment:
not related with this PR, but do we need to iterate, collect, and then sort? To get the lowest peer by that `Ord` criteria we can just iterate once and get the lowest with :
```rust
let new_peer = self
            .network_globals
            .peers
            .read()
            .synced_peers()
            .map(|peer| {
                (
                    failed_peers.contains(peer),
                    self.active_requests.get(peer).map(|v| v.len()).unwrap_or(0),
                    rand::thread_rng().gen::<u32>(),
                    *peer,
                )
            })
            .min()
            .map(|(_, _, _, peer)| peer);
```

same applies to the other change on this PR